### PR TITLE
Addressing issues with the new GenfacadesSourceTask

### PR DESCRIPTION
**Pull request by Anipik**
- moving using task to main file so that shims can use this task
- adding license header
- exposing genfacades assembly path variable if we manually want to set the path
- Added single quotes for generated file path

### CODE REVIEW

File: src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenFacades.targets [hunk 1]
```diff
@@ -6,6 +7,53 @@
...
   </PropertyGroup>
 
+  <PropertyGroup>
+    <GenFacadesTargetAssemblyPath Condition="'$(GenFacadesTargetAssemblyPath)' != ''">$(_MicrosoftDotNetGenFacadesTaskDir)Microsoft.DotNet.GenFacades.dll</GenFacadesTargetAssemblyPath>
+  </PropertyGroup>
+
+  <UsingTask TaskName="GenFacadesTask" AssemblyFile="$(GenFacadesTargetAssemblyPath)" />
+  <UsingTask TaskName="GenPartialFacadeSource" AssemblyFile="$(GenFacadesTargetAssemblyPath)" />
+
+  <PropertyGroup Condition="'$(IsPartialFacadeAssembly)' == 'true' AND '$(IsReferenceAssembly)' != 'true'">
```
Comment (danmosemsft) [inline src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenFacades.targets]: Nit, lowercase seems generally preferred for `and` and `or` in .targets/.props

Comment (ericstj) [inline src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenFacades.targets]: Why move all these property groups up?  It's resulting in a lot of duplicated conditions.

Comment (Anipik) [reply src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenFacades.targets]: I thought that in here https://github.com/dotnet/arcade/pull/2221#discussion_r270531112 
you meant to separate the properties and targets ?

Comment (ericstj) [reply src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenFacades.targets]: >  maybe some properties too, haven't done a deep dive

I was saying that to indicate that I wasn't sure if moving the task was enough, you should test to see if you needed anything else (or just review what else was used by the shims.proj).  The purpose of moving things up is to ensure it is defined when the conditions on the split targets were false, so copying those conditions really defeats the purpose...

File: src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenPartialFacadeSource.targets [hunk 2]
```diff
@@ -1,31 +1,36 @@
...
 <Project>
   <PropertyGroup>
     <!-- Tell ResolveMatchingContract to run and resolve contract to project reference -->
     <ResolveMatchingContract>true</ResolveMatchingContract>
     <GeneratePartialFacadeSourceDependsOn>
       $(GeneratePartialFacadeSourceDependsOn);ResolveMatchingContract
     </GeneratePartialFacadeSourceDependsOn>
+
+    <GeneratePartialFacadeSourceDependsOn Condition="'$(GeneratePlatformNotSupportedAssembly)' == 'true' or '$(GeneratePlatformNotSupportedAssemblyMessage)' != ''">
+      $(GeneratePartialFacadeSourceDependsOn);GenerateNotSupportedSource
```
Comment (ericstj) [inline src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenPartialFacadeSource.targets]: Don't use DependsOn for this.  MSBuild will fail if a target listed as DependsOn isn't present.  We don't require folks to use both GenFacades and GenAPI.  Can you come up with target sequencing that works well for both?

File: src/Microsoft.DotNet.GenAPI/build/Microsoft.DotNet.GenAPI.targets [hunk 3]
```diff
@@ -58,7 +58,7 @@
     <!-- Tell ResolveMatchingContract to run and resolve contract to project reference -->
     <ResolveMatchingContract>true</ResolveMatchingContract>
     <NotSupportedSourceFile>$(IntermediateOutputPath)$(TargetName).notsupported.cs</NotSupportedSourceFile>
-    <CoreCompileDependsOn>$(CoreCompileDependsOn);GenerateNotSupportedSource</CoreCompileDependsOn>
+    <CoreCompileDependsOn>GenerateNotSupportedSource;$(CoreCompileDependsOn)</CoreCompileDependsOn>
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.GenAPI/build/Microsoft.DotNet.GenAPI.targets]: What was the reason that you had to switch the order here?

File: src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenFacades.targets [hunk 4]
```diff
@@ -1,11 +1,18 @@
 <?xml version="1.0" encoding="utf-8"?>
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
 <Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenFacades.targets]: We don't need Tools Version and xmlns anymore. Also like 1 can be deleted. That applies to all msbuild files in the project.

**Comment by ViktorHofer** [comment]
LGTM

File: src/Microsoft.DotNet.GenAPI/build/Microsoft.DotNet.GenAPI.targets (hunk 3 above)
Comment (ericstj) [reply src/Microsoft.DotNet.GenAPI/build/Microsoft.DotNet.GenAPI.targets]: NotSupported assembly tries to fill entire contract.  GenFacades tries to fill contract.  I'm guessing he was hitting cases where we had both type forwards and typedefs for types.  I think this ordering is correct, or at least it matches what used to happen (genfacades running after compile and all).  If GenerateNotSupportedSource was smart enough to subset like GenFacades it probably wouldn't matter.  It's probably worth doing something like that in the future.  I know we've hit some cases in the past where we wanted "partial" not supported assemblies.

**Pull request merged**